boa_engine/src/builtins/intl/collator/options.rs

```diff
@@ -0,0 +1,83 @@
+use std::str::FromStr;
+
+use icu_collator::{CaseLevel, Strength};
+
+use crate::builtins::intl::options::OptionTypeParsable;
+
+#[derive(Debug, Clone, Copy)]
+#[repr(u8)]
```

Comment:
Why are we forcing the `u8` repr here?

Reply:
Just to reduce the overall size of `Collator`, since a `u8` repr gives the compiler more opportunities to rearrange the `Collator`'s fields to optimize for memory.

Reply:
I thought the compiler would do this automatically if needed, if there weren't many variants. OK, I see the point :)